### PR TITLE
[jsx-wrap-multilines] Add new nodes for wrap

### PR DESCRIPTION
**Pull request by evgeny-petukhov**
Add ConditionalExpression, LogicalExpression and JSXAttribute

### CODE REVIEW

File: docs/rules/jsx-wrap-multilines.md [hunk 1]
```diff
@@ -118,3 +121,62 @@ var hello = () => (
...
+      <p>Hello World</p>
+    </div>
+  }
+</div>
+```
+
+The following patterns are not considered warnings when configured `{logical: true}`.
+
+```jsx
+<div> not
```
Comment (ljharb) [inline docs/rules/jsx-wrap-multilines.md]: ?

File: tests/lib/rules/jsx-wrap-multilines.js [hunk 2]
```diff
@@ -201,20 +232,6 @@ ruleTester.run('jsx-wrap-multilines', rule, {
       options: [{return: true}],
       errors: [{message: 'Missing parentheses around multilines JSX'}]
     }, {
-      code: DECLARATION_TERNARY_NO_PAREN,
```
Comment (ljharb) [inline tests/lib/rules/jsx-wrap-multilines.js]: why do these tests need to be removed?

**Comment by alanevEugene** [comment]
Fix example for valid jsx in the attribute

File: tests/lib/rules/jsx-wrap-multilines.js (hunk 2 above)
Comment (alanevEugene) [inline tests/lib/rules/jsx-wrap-multilines.js]: This is the same tests as tests for condition rule. I can return them if you think that they are useful. The more tests the better

Comment (ljharb) [inline tests/lib/rules/jsx-wrap-multilines.js]: For semver-minor PRs, it makes me happier when tests are only added :-)

**Comment by alanevEugene** [comment]
Return tests.
Also I've added conditions to defaults. They were already part of `declaration` and `assignment` rules. Now it is separated and becomes more common rule.

**Comment by alanevEugene** [comment]
@ljharb Can you review again, please?

File: docs/rules/jsx-wrap-multilines.md [hunk 3]
```diff
@@ -6,14 +6,17 @@ Wrapping multiline JSX in parentheses can improve readability and/or convenience
...
 
 There are the possible syntax available:
 
 * `declaration`
 * `assignment`
 * `return`
 * `arrow`
+* `condition`
+* `logical` (not enabled by default)
+* `attr` (not enabled by default)
```
Comment (ljharb) [inline docs/rules/jsx-wrap-multilines.md]: this should probably be "prop" - in jsx, elements have props, and "attributes" are something HTML elements have (which jsx isn't)

File: tests/lib/rules/jsx-wrap-multilines.js [hunk 4]
```diff
@@ -155,14 +217,14 @@ ruleTester.run('jsx-wrap-multilines', rule, {
       code: DECLARATION_TERNARY_PAREN
     }, {
       code: DECLARATION_TERNARY_NO_PAREN,
-      options: [{declaration: false}]
+      options: [{condition: false}]
```
Comment (ljharb) [inline tests/lib/rules/jsx-wrap-multilines.js]: why does this option need to change?

File: tests/lib/rules/jsx-wrap-multilines.js [hunk 5]
```diff
@@ -155,14 +217,14 @@ ruleTester.run('jsx-wrap-multilines', rule, {
...
     }, {
       code: ASSIGNMENT_TERNARY_SINGLE_LINE
     }, {
       code: ASSIGNMENT_TERNARY_PAREN
     }, {
       code: ASSIGNMENT_TERNARY_NO_PAREN,
-      options: [{assignment: false}]
+      options: [{condition: false}]
```
Comment (ljharb) [inline tests/lib/rules/jsx-wrap-multilines.js]: why does this option need to change?

File: docs/rules/jsx-wrap-multilines.md (hunk 3 above)
Comment (alanevEugene) [reply docs/rules/jsx-wrap-multilines.md]: Agreed!
I've renamed to `prop`

File: tests/lib/rules/jsx-wrap-multilines.js (hunk 4 above)
Comment (alanevEugene) [reply tests/lib/rules/jsx-wrap-multilines.js]: This test case checked the condition in declaration. Now checking the condition is separated.

File: tests/lib/rules/jsx-wrap-multilines.js (hunk 5 above)
Comment (alanevEugene) [reply tests/lib/rules/jsx-wrap-multilines.js]: This test case checked the condition in assignment. Now checking the condition is separated.

File: tests/lib/rules/jsx-wrap-multilines.js (hunk 4 above)
Comment (ljharb) [reply tests/lib/rules/jsx-wrap-multilines.js]: Does that mean that "declaration: true" now doesn't check as much as it used to? That would make this a breaking change.

File: tests/lib/rules/jsx-wrap-multilines.js (hunk 5 above)
Comment (ljharb) [reply tests/lib/rules/jsx-wrap-multilines.js]: Similarly, does that mean that "assignment: true" now doesn't check as much as it used to? That would make this a breaking change.

File: tests/lib/rules/jsx-wrap-multilines.js (hunk 4 above)
Comment (alanevEugene) [reply tests/lib/rules/jsx-wrap-multilines.js]: Yes, it is a broken change for declarations. But now the logic of checking is better

File: tests/lib/rules/jsx-wrap-multilines.js (hunk 5 above)
Comment (alanevEugene) [reply tests/lib/rules/jsx-wrap-multilines.js]: Yes, it is a broken change for assignments too

File: tests/lib/rules/jsx-wrap-multilines.js (hunk 4 above)
Comment (ljharb) [reply tests/lib/rules/jsx-wrap-multilines.js]: Is there any way we could avoid making it breaking?

Like if `condition` is absent, it uses the old behavior, but if it's present (true or false) then it uses the new behavior?

Comment (alanevEugene) [reply tests/lib/rules/jsx-wrap-multilines.js]: Yes, I can do it now. But it is not very good, so I suggest to do this only as temporary solution, and separate checking condition in major release

Comment (ljharb) [reply tests/lib/rules/jsx-wrap-multilines.js]: Yes, definitely we'd want to separate that in the next major - but for now, we'd be able to add these new options as semver-minor.

**Comment by alanevEugene** [comment]
Now there are only additions, so this version is fully compatible with previous.

**Comment by alanevEugene** [comment]
@ljharb Is it would be useful to add warning?

**Comment by ljharb** [comment]
What kind of warning?

**Comment by alanevEugene** [comment]
To enable checking condition if they now use conditions in declaration or assignment

**Comment by alanevEugene** [comment]
@ljharb Hello! What status of this PR? Do you need these changes in the project?

**Comment by ljharb** [comment]
I'm waiting for reviews from other collaborators.

**Pull request merged**